internal/core/application/operator_service.go

```diff
@@ -1596,6 +1602,67 @@ func (o *operatorService) ListWebhooks(
 	return hooks, nil
 }
 
+func (o *operatorService) GetMarketReport(
+	ctx context.Context,
+	market Market,
+	timeRange TimeRange,
+) (*MarketReport, error) {
+	startTime, endTime, err := timeRange.getStartAndEndTime(time.Now())
+	if err != nil {
+		return nil, err
+	}
+
+	m, _, err := o.repoManager.MarketRepository().GetMarketByAssets(
+		ctx, market.BaseAsset, market.QuoteAsset,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if m == nil {
+		return nil, ErrMarketNotExist
+	}
+
+	trades, err := o.repoManager.TradeRepository().GetCompletedTradesByMarket(
+		ctx, market.QuoteAsset,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	//sort desc
+	sort.SliceStable(trades, func(i, j int) bool {
+		return trades[i].SwapRequest.Timestamp > trades[j].SwapRequest.Timestamp
+	})
+
+	totalFees := make(map[string]int64)
+	volume := make(map[string]int64)
+	for _, trade := range trades {
+		if time.Unix(int64(trade.SwapRequest.Timestamp), 0).After(startTime) &&
+			time.Unix(int64(trade.SwapRequest.Timestamp), 0).Before(endTime) {
```

Comment:
this way we exclude trades with timestamp equal to either startTime and endTime, which might be ok but we must make this explicit at spec level

Reply:
Done